framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,324 @@
+     *     the remainder from dividing an arbitrary value in this range by an arbitrary value
+     *     in @param
+     */
+    public Range remainder(Range right) {
+        long[] possibleValues = new long[9];
+        possibleValues[0] = 0;
+        possibleValues[1] = Math.min(from, Math.abs(right.from) - 1);
+        possibleValues[2] = Math.min(from, Math.abs(right.to) - 1);
+        possibleValues[3] = Math.min(to, Math.abs(right.from) - 1);
+        possibleValues[4] = Math.min(to, Math.abs(right.to) - 1);
+        possibleValues[5] = Math.max(from, -Math.abs(right.from) + 1);
+        possibleValues[6] = Math.max(from, -Math.abs(right.to) + 1);
+        possibleValues[7] = Math.max(to, -Math.abs(right.from) + 1);
+        possibleValues[8] = Math.max(to, -Math.abs(right.to) + 1);
+        return getRangeFromPossibleValues(possibleValues);
+    }
+
+    /**
+     * Left shifts a range by another
+     *
+     * @param right the range of bits to be shifted
+     * @return the range of the resulted value from left shifting an arbitrary value in this range
+     *     by an arbitrary value in @param.
+     */
+    public Range shiftLeft(Range right) {
+        // TODO: warning if right operand may be out of [0, 31]
+        if (right.from < 0 || right.from > 31 || right.to < 0 || right.to > 31) {
+            return new Range();
+        }
+        long resultFrom = from << (from >= 0 ? right.from : right.to);
+        long resultTo = to << (to >= 0 ? right.to : right.from);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Signed right shifts a range by another
+     *
+     * @param right the range of bits to be shifted
+     * @return the range of the resulted value from signed right shifting an arbitrary value in this
+     *     range by an arbitrary value in @param.
+     */
+    public Range signedShiftRight(Range right) {
+        if (right.from < 0 || right.from > 31 || right.to < 0 || right.to > 31) {
+            return new Range();
+        }
+        long resultFrom = from >> (from >= 0 ? right.to : right.from);
+        long resultTo = to >> (to >= 0 ? right.from : right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Unary plus this range
+     *
+     * @return the resulted range of applying unary plus on an arbitrary value in this range
+     */
+    public Range unaryPlus() {
+        return new Range(from, to);
+    }
+
+    /**
+     * Unary minus this range
+     *
+     * @return the resulted range of applying unary minus on an arbitrary value in this range
+     */
+    public Range unaryMinus() {
+        return new Range(-to, -from);
+    }
+
+    /**
+     * Bitwise complements this range
+     *
+     * @return the resulted range of applying bitwise complement on an arbitrary value in this range
+     */
+    public Range bitwiseComplement() {
+        return new Range(~to, ~from);
+    }
+
+    /**
+     * Control flow refinement for less than operator
```

Comment:
This comment documents how the method is used, but not what the method means.  Can you first say what the method means?  Or, can you give a code example to show how it is used?  Otherwise, it's hard for a reader to understand whether the code is correct.